File: backend/VehicleServiceAPI/Models/Vehicle.cs
using System.Collections.Generic;

namespace VehicleServiceAPI.Models
{
    public class Vehicle
    {
        public int Id { get; set; }
        public int CustomerId { get; set; }
        public string Make { get; set; }
        public string Model { get; set; }
        public int Year { get; set; }
        public string LicensePlate { get; set; }
        public string Color { get; set; }
        public bool IsDeleted { get; set; }

        public User Customer { get; set; }
        public ICollection<ServiceRequest> ServiceRequests { get; set; }
    }
}

File: frontend/src/types/vehicle.ts
export interface Vehicle {
  id: number;
  customerId: number;
  customerName?: string;
  make: string;
  model: string;
  year: number;
  licensePlate: string;
  color?: string;
}

export interface CreateVehicleRequest {
  make: string;
  model: string;
  year: number;
  licensePlate: string;
  color?: string;
}
